src/Raven.Server/Documents/PeriodicBackup/BackupHelper.cs

```diff
@@ -29,5 +34,40 @@ public static bool BackupTypeChanged(PeriodicBackupConfiguration previous, Perio
         {
             return previous.BackupType != current.BackupType;
         }
+
+        public static async Task<long> RunWithRetriesAsync(
+            int maxRetries,
+            Func<Task<long>> action,
+            string infoMessage,
+            string errorMessage,
+            SmugglerResult smugglerResult,
+            Action<IOperationProgress> onProgress = default,
+            OperationCancelToken operationCancelToken = default)
+        {
+            var retries = 0;
+
+            while (true)
+            {
+                try
+                {
+                    if (retries > 0)
+                        operationCancelToken?.Token.ThrowIfCancellationRequested();
+
+                    smugglerResult?.AddInfo(infoMessage);
+                    onProgress?.Invoke(smugglerResult?.Progress);
+
+                    return await action();
+                }
+                catch (TimeoutException)
+                {
+                    if (++retries < maxRetries)
```

Comment:
shouldn't there be any throttling between attempts e.g. 100ms ?

Reply:
Done.